homeassistant/components/huawei_lte/services.yaml

```diff
@@ -12,6 +12,40 @@ reboot:
       description: URL of router to reboot; optional when only one is configured.
       example: http://192.168.100.1/
 
+net_mode:
+  description: Select network mode and bands.
+  fields:
+    networkmode:
+      description: Preferred network mode.
+      example: '"00" for 4G/3G/2G, "01" for 2G, "02" for 3G, "03" for 4G, "0301" for 4G/2G, "0302" for 4G/3G, "0201" for 3G/2G.'
+    networkband:
+      description: UMTS and GSM Bands; optional.
+      example: 3FFFFFFF
+    lteband:
+      description: LTE Bands; optional.
+      example: 7FFFFFFFFFFFFFFF
+    url:
+      description: URL of router; optional when only one is configured.
+      example: http://192.168.100.1/
+
+register:
+  description: Connect to a specific operator.
+  fields:
+    mode:
+      description: Operator search mode.
+      example: '"0" for automatic, "1" for manual selection.'
+    plmn:
+      description: >
+        Operator plmn to connect to; leave empty if mode is '0'. PLMN: 5 or 6 digit number
+        consisting of Mobile Country Code (MCC) and Mobile Network Code (MNC).
+      example: 310160
+    rat:
+      description: 'Band connect to; leave empty if mode is "0".'
+      example: '"0" for 2G, "2" for 3G, "7" for 4G.'
```

Comment:
I think instead of documenting the somewhat cryptic "Use 0 for 2G" etc possible values, we should do that in code instead, so the HA user would enter `2G`/`3G`/`4G` for this and we'd do the corresponding mapping in code before calling the API. We should also amend that so that one can also call them with e.g. `raw:0` or something like that syntax in order to send raw API values as-is (0 in this example) so that it is still usable even if we don't have a mapping for some values.

Applies to all values in all these register and net_mode calls, not just register->rat.

BTW I have no idea what the example values for networkband and lteband mean. But if we can assign a user friendlier value scheme for them per the above (maybe comma separated if the bands take multiple values, dunno), that becomes easier.